spec/index_spec.rb

```diff
@@ -372,4 +372,51 @@
       )
     end
   end
+
+  describe Daru::CategoricalIndex do
+    before do
+      @idx = Daru::CategoricalIndex.new(
```

Comment:
I strongly propose to make it "right" from the beginning of new specs and use [subject](http://betterspecs.org/#subject) and [let](http://betterspecs.org/#let), with readable names of subjects/lets. Let the old spec have instance variables for now, but new ones should use modern RSpec features. @v0dro, WDYT?

Also, I think it is better to have new large classes/features specs in separate files.